pkg/templates/k8sresource_funcs.go

```diff
@@ -81,3 +94,55 @@ func base64decode(v string) string {
 
 	return string(data)
 }
+
+// protect encrypts the input value using AES-CBC. If a salt is set on t.config.Salt, it will prefix the plaintext
```

Comment:
The salt should be randomly generated and not user supplied. Allowing the user to determine the salt is likely to lead to salt re-use which is a common mistake in proper use of salts in cryptography.
https://en.wikipedia.org/wiki/Salt_(cryptography)#Salt_re-use